airbyte-integrations/bases/base-normalization/dbt-project-template/macros/cross_db_utils/surrogate_key.sql

```diff
@@ -11,3 +11,20 @@
         {%- endfor %}
     )
 {%- endmacro %}
+
+{# MSSQL surrogate_key #}
+{%- macro sqlserver__surrogate_key(field_list) -%}
+    {% set fields = [] %}
+    {%- for field in field_list -%}
+        {% set _ = fields.append("coalesce(cast(" ~ field ~ " as " ~ dbt_utils.type_string() ~ "), '')") %}
+    {%- endfor -%}
+
+    {#-- CONCAT() in SQL SERVER accepts from 2 to 254 arguments, we use batches for the main concat, to overcome the limit. --#}
+    {% set concat_chunks = [] %}
+    {% for chunk in fields|batch(253) -%}
+        {% set _ = concat_chunks.append(dbt_utils.concat(chunk)) %}
+    {% endfor %}
+
+    {#-- Apply the main concat() to the chunked concats. --#}
+    {{dbt_utils.hash(dbt_utils.concat(concat_chunks))}}
```

Comment:
Can this specific implementation of concat function because of SQL Server limitation be moved to the concat macro instead of surrogate_key?

Reply:
Good point), moved it there.